openshift-kube-apiserver/authorization/minimumkubeletversion/minimum_kubelet_version.go

```diff
@@ -0,0 +1,124 @@
+package minimumkubeletversion
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/blang/semver/v4"
+	nodelib "github.com/openshift/library-go/pkg/config/node"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/auth/nodeidentifier"
+)
+
+var (
+	minimumKubeletVersion *semver.Version
+	versionGotten         bool
+	versionSet            bool
+	versionLock           sync.Mutex
+)
+
+// GetMinimumKubeletVersion retrieves the set global minimum kubelet version in a safe way.
+// It ensures it is only retrieved once, and is set before it's retrieved.
+// It will panic on any error.
+// The global value should only be gotten through this function.
+func GetMinimumKubeletVersion() *semver.Version {
+	versionLock.Lock()
+	defer versionLock.Unlock()
+	if versionGotten {
+		panic("coding error: MinimumKubeletVersion already gotten")
+	}
```

Comment:
This case seems safe, and you might have issues with this in integration tests that start multiple apiservers or a unit test that creates multiple authorizers.

Reply:
dropped the field entirely